dashboard/app/models/unit_group.rb

```diff
@@ -164,12 +164,22 @@ def write_serialization
   def update_scripts(new_scripts, alternate_scripts = nil)
     alternate_scripts ||= []
     new_scripts = new_scripts.reject(&:empty?)
+    new_scripts_objects = new_scripts.map {|s| Script.find_by_name!(s)}
     # we want to delete existing unit group units that aren't in our new list
-    scripts_to_delete = default_unit_group_units.map(&:script).map(&:name) - new_scripts
-    scripts_to_delete -= alternate_scripts.map {|hash| hash['alternate_script']}
+    scripts_to_delete = default_unit_group_units.map(&:script) - new_scripts_objects
+    scripts_to_delete -= alternate_scripts.map {|hash| Script.find_by_name!(hash['alternate_script'])}
 
-    new_scripts.each_with_index do |script_name, index|
-      script = Script.find_by_name!(script_name)
+    if scripts_to_delete.any?(&:prevent_course_version_change?)
```

Comment:
is there a reason we want this logic to live here, rather than in, say, a `before_destroy` hook on Script?

Reply:
The variables could be better named but this isn't actually destroying a script, just removing it from the unit group.

Reply:
ahhhhh, right. Yeah, something like `scripts_to_remove` would be clearer

Reply:
Agreed -- I'll make that change